### PR TITLE
Fixed Sorting Functions to new-style key where applicable

### PR DESCRIPTION
**Pull request by maweki**
I hopefully translated all instances of `.sort` that would error out correctly.

### CODE REVIEW

**Comment by maweki** [comment]
Sadly, this doesn't fix sorting in the main recipie view.

File: gourmet/gtk_extras/pageable_store.py [hunk 1]
```diff
@@ -167,17 +167,13 @@ def sort (self, column, direction=FORWARD):
...
         if direction==self.OFF:
             self.parent_list = self.unsorted_parent
             return
-        if direction==self.FORWARD: shift_by = 1
-        elif direction==self.REVERSE: shift_by = -1
-        self.parent_list.sort(lambda r1,r2: ((r1[column]>r2[column] and 1*shift_by) or
-                                             (r1[column]<r2[column] and -1*shift_by) or
-                                             0)
-                              )
+        self.parent_list.sort(key=lambda x: x[column], reverse=(direction == self.REVERSE))
```
Comment (maweki) [inline gourmet/gtk_extras/pageable_store.py:176]: I think the reversal condition is wrong, but the original looks wrong in that case

File: gourmet/keymanager.py [hunk 2]
```diff
@@ -201,7 +201,7 @@ def look_for_key (self, txt):
                 # Add some probability if our word shows up in the key
                 if ik.find(w)>=0: retvals[ik]+=0.1
         retv = list(retvals.items())
-        retv.sort(lambda a,b: a[1]<b[1] and 1 or a[1]>b[1] and -1 or 0)
+        retv.sort(key=lambda x: x[1])
```
Comment (maweki) [inline gourmet/keymanager.py:204]: I think this maybe needs to be reversed as well

File: gourmet/plugins/import_export/mealmaster_plugin/mealmaster_importer.py [hunk 3]
```diff
@@ -539,8 +539,8 @@ def find_columns (strings, char=" "):
     # we start with the columns in the first string
     if not strings:
         return None
-    strings=strings[0:]
-    strings.sort(lambda x,y: len(x)>len(y))
+    strings=list(strings)
+    strings.sort(key=len, reverse=True)
```
Comment (maweki) [inline gourmet/plugins/import_export/mealmaster_plugin/mealmaster_importer.py:543]: I'm not sure about this one either, as the original doesn't even return -1, 0, 1

File: gourmet/importers/importManager.py [hunk 4]
```diff
@@ -222,8 +222,7 @@ def get_tempfilename (self, url, data, content_type):
     def guess_extension (self, content_type):
         if content_type in self.extensions_by_mimetype:
             answers = list(self.extensions_by_mimetype[content_type].items())
-            answers.sort(lambda a,b: cmp(a[1],b[1])) # sort by count...
-            return answers[-1][0] # Return the most frequent
+            return max(answers, key=lambda x: x[1])[0] # Return the most frequent by count...
```
Comment (cydanil) [inline gourmet/importers/importManager.py:225]: much clearer indeed

**Review by cydanil — APPROVED** [review]
Thanks for taking the time to look at these!

Some of the changes require more knowledge, such as in `pageable_store.py`, so  I generally approve, but I haven't made the effort to poke around.

**Pull request merged**